cmake/platform/osx/osx.cmake

```diff
@@ -18,3 +18,4 @@ elseif(APP_WINDOW_SYSTEM STREQUAL native)
 else()
   message(SEND_ERROR "Only SDL or native windowing options are supported.")
 endif()
+list(APPEND PLATFORM_OPTIONAL_DEPS Smctemp)
```

Comment:
It's not really optional, as you don't do ifdef checks in core code.  Change to PLATFORM_REQUIRED_DEPS